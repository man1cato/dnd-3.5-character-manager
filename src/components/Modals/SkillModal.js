import React from 'react'

import DefaultModal from './DefaultModal'
import { convertTextToArray } from '../../utils/utils'


const SkillModal = ({clickedSkill, handleCloseModal}) => (
    <DefaultModal
        clickedItem={clickedSkill}
        onRequestClose={handleCloseModal}
        contentLabel="Selected Skill"
    >
        {clickedSkill && 
            <>
                <h4>{clickedSkill.name}</h4>
                <h5>{clickedSkill.keyAbility}</h5>
                {convertTextToArray(clickedSkill.description).map((paragraph, i) => 
                    <p key={i}>{paragraph}</p>
                )}
                {clickedSkill.notes && <p>Notes: {clickedSkill.notes}</p>}
            </>
        }
    </DefaultModal>
)

export default SkillModal