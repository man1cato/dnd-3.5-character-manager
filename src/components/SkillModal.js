import React from 'react';
import Modal from 'react-modal';

Modal.setAppElement('#app');

const SkillModal = ({selectedSkill, handleCloseModal}) => (
    <Modal
        isOpen={!!selectedSkill}
        onRequestClose={handleCloseModal}
        contentLabel="Selected Skill"
    >
        {selectedSkill && 
            <div>
                <h4>{selectedSkill.name}</h4>
                <h5>{selectedSkill.keyAbility}</h5>
                <p>{selectedSkill.description}</p>
                {selectedSkill.notes && <p>Notes: {selectedSkill.notes}</p>}
            </div>
        }
    </Modal>

)

export default SkillModal;