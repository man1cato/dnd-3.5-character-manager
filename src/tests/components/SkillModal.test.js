import React from 'react'
import {shallow} from 'enzyme'
import fs from 'fs'

import SkillModal from '../../components/SkillModal'


let skills, props

beforeAll(async () => {
   const api = await fs.promises.readFile('src/tests/fixtures/api.json')
   skills = JSON.parse(api).skills
   props = {
      selected: Object.values(skills)[0]
   }
})


test('should render SkillModal', () => {
   const wrapper = shallow(<SkillModal {...props} />)
   expect(wrapper).toMatchSnapshot()
})