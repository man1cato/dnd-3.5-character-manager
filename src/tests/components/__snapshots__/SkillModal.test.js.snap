// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SkillModal 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={0}
  contentLabel="Selected Skill"
  isOpen={true}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <div>
    <h4>
      Knowledge (Local)
    </h4>
    <h5>
      INT
    </h5>
    <p
      key="0"
    >
      You possess knowledge of local customs and events. 
    </p>
  </div>
</Modal>
`;
